### ISSUE TITLE
Improve the KMP SDK

### ISSUE THREAD

**Issue by buenaflor** [getsentry/sentry-kotlin-multiplatform#195]
### Description

Meta issue for tracking the most important improvements we can do for the SDK currently

```[tasklist]
### Tasks
- [ ] https://github.com/getsentry/sentry-kotlin-multiplatform/issues/74
- [ ] https://github.com/getsentry/sentry-kotlin-multiplatform/issues/108
- [ ] https://github.com/getsentry/sentry-kotlin-multiplatform/issues/191
- [ ] https://github.com/getsentry/sentry-kotlin-multiplatform/issues/174
```

```[tasklist]
### Docs
- [ ] Update getting started docs to use Gradle Plugin by default, keep current version as separate page
```

```[tasklist]
### Optional / nice-to-have / later
- [x] ~Gradle plugin: upload proguard & dsyms directly from the gradle plugin~ This is blocked by the SAGP multi module rework
- [ ] Refactor KMP plugin to work on root project level for multi module apps
```

**Comment by buenaflor** [comment]
Fixes have been implemented through the gradle plugin
